### PR TITLE
use slim image

### PR DESCRIPTION
**Pull request by machisuji**
WP [#50645](https://community.openproject.org/projects/devops/work_packages/50645/activity)

This PR uses the `-slim` tag for the used images. To make that work the following things were done:

- [x] create new deployment for worker process
- [x] create batch job for migrating and seeding the database
- [x] introduce init containers for deployments to wait for the database

Depends on https://github.com/opf/openproject/pull/13826.

### CODE REVIEW

File: charts/openproject/templates/seeder-job.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "common.names.fullname" . }}-seeder-{{ now | date "20060102150405" }}
```
Comment (machisuji) [inline charts/openproject/templates/seeder-job.yaml:4]: We put the current time in the job name so we don't need to work with hooks to re-run the seeder on each upgrade.

**Comment by machisuji** [comment]
On OpenShift we still have the problem that the web process cannot start because it doesn't have the necessary permissions to create a tmp directory.

```
=> Booting Puma
=> Rails 7.0.7 application starting in production 
=> Run `bin/rails server --help` for more startup options
Exiting
/usr/local/lib/ruby/3.2.0/fileutils.rb:406:in `mkdir': Permission denied @ dir_s_mkdir - /app/tmp (Errno::EACCES)
        from /usr/local/lib/ruby/3.2.0/fileutils.rb:406:in `fu_mkdir'
```

This is because by default on OpenShift you get assigned a uid and you can't request ours (1000 for `app`) without extra privileges. This id might be something like `1004560000` but all the files in the container belong to `app`, that is id `1000`.

The question is how to solve this. As far as I could find out you can't get rails to create the tmp folder elsewhere. Perhaps the simplest solution would be to have the container build create the tmp directory with global write permissions.

But that wouldn't be pretty. I'm not sure how bitnami postgres container does it but they somehow manage to change the owner of the relevant files to the assigned uid in an [init container](https://github.com/bitnami/charts/blob/main/bitnami/postgresql/templates/primary/statefulset.yaml#L111-L148). 

```
1004560000@openproject-dev-postgresql-0:/$ ls -lisah /bitnami/postgresql/data/
total 96K
393217 4.0K drwx--S---. 19 1004560000 1004560000 4.0K Sep 28 17:27 .
     2 4.0K drwxrwsr-x.  4 root       1004560000 4.0K Sep 25 09:25 ..
393231 4.0K drwxrws---.  6 1004560000 1004560000 4.0K Sep 25 09:25 base
393219 4.0K drwxrws---.  2 1004560000 1004560000 4.0K Sep 28 17:31 global
393221 4.0K drwxrws---.  2 1004560000 1004560000 4.0K Sep 25 09:25 pg_commit_ts
393222 4.0K drwxrws---.  2 1004560000 1004560000 4.0K Sep 25 09:25 pg_dynshmem
393245 4.0K -rw-rw----.  1 1004560000 1004560000 1.6K Sep 25 09:25 pg_ident.conf
```

Maybe we can use root in init containers to do that? Will investigate.

**Comment by machisuji** [comment]
On another note the install test in the workflow is failing although it looks fine. All the pods start correctly. Will have to look into that tomorrow as well.

**Review by oliverguenther — APPROVED** [review]

**Comment by machisuji** [comment]
I do realise there is a whole lot of duplication in there still. We need to put this into a template at some point.

**Comment by machisuji** [comment]
In any case this should now be good to use.

**Pull request merged**